app/helpers/blacklight/facets_helper_behavior.rb

```diff
@@ -136,20 +136,28 @@ def render_facet_value(facet_field, item, options ={})
   def path_for_facet(facet_field, item)
     facet_config = facet_configuration_for_field(facet_field)
     if facet_config.url_method
-      path = send(facet_config.url_method, facet_field, item)
+      send(facet_config.url_method, facet_field, item)
     else
-      path = search_action_path(add_facet_params_and_redirect(facet_field, item))
+      search_action_path(current_path.add_facet_params_and_redirect(facet_field, item))
     end
   end
 
+  def current_path
```

Comment:
I'm fine with moving it to `UrlHelperBehavior` I put it here, because this is where all that functionality was extracted from.

It can be memoized, there is not reason not to.

I'll accept any name you suggest, this was the best I had.